google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -1287,6 +1287,20 @@ def create_new_session
           Session.from_grpc grpc, @project.service
         end
 
+        ##
+        # @private
+        # Creates a batch of new session objects of size `session_count`.
+        def batch_create_new_sessions session_count
+          ensure_service!
+          resp = @project.service.batch_create_sessions \
+            Admin::Database::V1::DatabaseAdminClient.database_path(
+              project_id, instance_id, database_id
+            ),
+            session_count,
+            labels: @session_labels
+          resp.session.map { |grpc| Session.from_grpc grpc, @project.service }
```

Comment:
also if possible, use the most recently used session again for the subsequent requests.. 
therefore the pool should be a LIFO data structure (like stack).

this will make sure the sessions are used from backend caches (and keep the cache hit ratio high), which will reduce the read / write latencies on user requests.

Reply:
OK, I will change the way sessions are checked out from the pool in order to get the desired LIFO behavior.

Reply:
Done.